cookbooks/ambari/recipes/default.rb

```diff
@@ -0,0 +1,28 @@
+#
+# Cookbook Name:: ambari-views-chef
+# Recipe:: default
+#
+# Copyright (c) 2016 Artem Ervits, All Rights Reserved.
+
+# include_recipe 'ambari::setattr'
+# dependencies
+%w(openssh-client wget curl unzip tar python2.7 openssl libpq5 ssl-cert mysql-client).each do |pkg|
+  package pkg do
+  end
+end
+
+case node['platform']
+when 'ubuntu'
+    apt_repository 'ambari' do
+      uri node['ambari']['ambari_ubuntu_repo_url']
+      components ['main']
+      distribution 'Ambari'
+      action :add
+      keyserver 'keyserver.ubuntu.com'
```

Comment:
Please allow `keyserver` and `key` to be optionally passed in; then we can take advantage of the dual-nature of the `key` attribute:
```If a keyserver is provided, this is assumed to be the fingerprint; otherwise it can be either the URI to the GPG key for the repo, or a cookbook_file. Default value: nil.``` - From the [Chef 12.15 Docs](https://github.com/chef/chef-web-docs/blob/2cdc74c9bbe78db723758715f60ba0b5eb46d394/chef_master/source/resource_apt_repository.rst)